drivers/sensor/lsm303dlhc_magneto/lsm303dlhc_magneto.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2018 Philémon Jaermann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SENSOR_LSM303DLHC_MAGN_
+#define __SENSOR_LSM303DLHC_MAGN_
+
+#define LSM303DLHC_MAGN_I2C_ADDRESS	CONFIG_LSM303DLHC_MAGN_I2C_ADDR
+
+#define LSM303_DLHC_MAGN_X_EN_BIT	BIT(0)
+#define LSM303DLHC_MAGN_Y_EN_BIT	BIT(1)
+#define LSM303DLHC_MAGN_Z_EN_BIT	BIT(2)
+#define LSM303DLHC_MAGN_EN_BITS		(LSM303_DLHC_MAGN_X_EN_BIT | \
+					LSM303DLHC_MAGN_Y_EN_BIT | \
+					LSM303DLHC_MAGN_Z_EN_BIT)
+
+#if defined(CONFIG_LSM303DLHC_MAGN_ODR_1)
+	#define LSM303DLHC_MAGN_ODR_IDX	1
```

Comment:
Data rate output bits are 3 with value between 0 to 7. LSM303DLHC_MAGN_ODR_IDX should begin from 0 and end to 7.